test/byContract/IexecERC20/IexecERC20.ts

```diff
@@ -0,0 +1,269 @@
+// SPDX-FileCopyrightText: 2020-2024 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
+// SPDX-License-Identifier: Apache-2.0
+
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { BigNumber } from 'ethers';
+import { ethers, expect } from 'hardhat';
+import { loadHardhatFixtureDeployment } from '../../../scripts/hardhat-fixture-deployer';
+import {
+    IexecInterfaceNative,
+    IexecInterfaceNative__factory,
+    TestReceiver,
+    TestReceiver__factory,
+} from '../../../typechain';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+import { IexecWrapper } from '../../utils/IexecWrapper';
+
+const value = 100;
+const zeroAddress = ethers.constants.AddressZero;
+
+describe('ERC20', async () => {
+    let proxyAddress: string;
+    let iexecWrapper: IexecWrapper;
+    let [iexecPoco, iexecPocoAsHolder, iexecPocoAsSpender]: IexecInterfaceNative[] = [];
+    let [holder, recipient, spender, anyone, zeroAddressSigner]: SignerWithAddress[] = [];
+    let totalSupplyBeforeFirstDeposit: BigNumber;
+
+    beforeEach('Deploy', async () => {
+        // Deploy all contracts
+        proxyAddress = await loadHardhatFixtureDeployment();
+        // Initialize test environment
+        await loadFixture(initFixture);
+    });
+
+    async function initFixture() {
+        const accounts = await getIexecAccounts();
+        // Setup current test accounts from some arbitrary iExec accounts
+        ({ requester: holder, beneficiary: recipient, anyone } = accounts);
+        spender = recipient;
+        zeroAddressSigner = await ethers.getImpersonatedSigner(zeroAddress);
+        iexecWrapper = new IexecWrapper(proxyAddress, accounts);
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+        iexecPocoAsHolder = iexecPoco.connect(holder);
+        iexecPocoAsSpender = iexecPoco.connect(spender);
+        totalSupplyBeforeFirstDeposit = await iexecPoco.totalSupply();
+        await iexecWrapper.depositInIexecAccount(holder, value);
+    }
+
+    describe('Total supply', () => {
+        it('Should get total supply', async () => {
+            expect(await iexecPoco.totalSupply()).equal(totalSupplyBeforeFirstDeposit.add(value));
+        });
```

Comment:
This is also tested in `IexecAccessors`. Let's choose one of them.

Reply:
I would say both tests makes sense, here for instance testing https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/IERC20.sol